app/reflexes/admin/connected_app_reflex.rb

```diff
@@ -7,10 +7,20 @@ def create
       authorize! :admin, enterprise
       app = ConnectedApp.create!(enterprise_id: enterprise.id)
 
+      selector = "#edit_enterprise_#{enterprise.id} #connected-app-discover-regen"
+      html = ApplicationController.render(
+        partial: "admin/enterprises/form/connected_apps",
+        locals: { enterprise: },
+      )
+
+      # Avoid race condition by sending before enqueuing job:
+      cable_ready.morph(selector:, html:).broadcast
```

Comment:
The change on the page is almost instant now. Previously, it was calling the edit action of the enterprises controller and there was a noticeable lag. That action renders lots of data, including a select box with all enterprises of the platform and all kind of options for every tab of the enterprise settings. In contrast, the partial is very small.